### ISSUE TITLE
Build ffmpeg with libx265 enabled

### ISSUE THREAD

**Issue by pkoppise** [Javernaut/ffmpeg-android-maker#79]
Hi 
I am trying to build ffmpeg for android plartform with libx265 enabled
Could you please help  me on that

**Comment by Javernaut** [comment]
Hello. Are you trying to integrate the library into this repo's scripts? At what state are you? Any specific problems you are facing?

**Comment by pkoppise** [comment]
Hi Yes Basically under scripts folder  i am trying to write scripts (download and compile for)  for libx265 similar to libx264.

Download script 
==============
#!/usr/bin/env bash

source ${SCRIPTS_DIR}/common-functions.sh

#version 3.5
LIBX265_VERSION=3bd3dd731b4b4c3fbbe5e513c16bc6ae481a0ec5

downloadTarArchive \
  "libx265" \
  "https://bitbucket.org/multicoreware/x265_git/get/${LIBX265_VERSION}.tar.gz"
  
  Build script 
  =========
  I am referring to this https://github.com/kimsan0622/libx265-android , able to build the library with some changes
  
  but i am getting compatibility issues. 
  ld: error: /home/phani/ffmpeg-android-maker/build/external/arm64-v8a/lib/libx265.a(api.cpp.o) is incompatible with aarch64linux

**Comment by Javernaut** [comment]
@pkoppise 
I have some progress with libx265. [This branch](https://github.com/Javernaut/ffmpeg-android-maker/tree/adding/libx265) contains a **very** raw implementation of the downloading and building, but currently only armv7 and x86_64 builds are [successful](https://github.com/Javernaut/ffmpeg-android-maker/actions/runs/6747639101). I'll be quite busy during next days, so just letting you know about the progress. You can pick it from here, if you want.

**Comment by pkoppise** [comment]
Hi @Javernaut 
Thanks for sharing the details , will go through the scripts and will try to do it for arm64-v8a. I will update you on the progress.

**Comment by Javernaut** [comment]
@pkoppise Any news from your side? I probably will have some time for the library soon.

**Comment by pkoppise** [comment]
hi @Javernaut 
I tried building for arm64-v8a platform but lot of errors came 

/home/phani/ffmpeg-android-maker/sources/libx265/x265_3.5/source/common/aarch64/sad-a.S:105:1: note: while in macro instantiation
SAD_X_8xN 4 32
^
<instantiation>:11:5: error: invalid instruction mnemonic 'uaddlv'
    uaddlv s3, v19.8h
    ^~~~~~
/home/phani/ffmpeg-android-maker/sources/libx265/x265_3.5/source/common/aarch64/sad-a.S:105:1: note: while in macro instantiation
SAD_X_8xN 4 32
^
<instantiation>:18:5: error: invalid instruction mnemonic 'stp'
    stp s0, s1, [x6]


Thought of looking more into it..could not find time..if you have time to look into it that would be really great.

**Comment by Javernaut** [comment]
![image](https://github.com/Javernaut/ffmpeg-android-maker/assets/979018/ac20cdb1-df25-4dc4-bf08-44d386f37602)

So, the library seems to compile fine now. The [latest release](https://github.com/Javernaut/ffmpeg-android-maker/releases/tag/v2.9) of the ffmpeg-android-maker has it integrated. I had to disable certain features of the x265 library to make it compilable. The [ffmpeg-kit](https://github.com/arthenica/ffmpeg-kit), for example, does the same. So I assume this is an acceptable solution. Check it out if some additional configuration is relevant to you.

And I found out that the libx265 contains only an encoder for HEVC, so I can't easily test it on my end. Will leave that to you.
